### ISSUE TITLE
Incorrect type inference of power expression with non-decimal exponent

### ISSUE THREAD

**Issue by qexat** [microsoft/pyright#4754]
**Describe the bug**
Assigning a power expression with an exponent written with any of the `0bX`/`0oX`/`0xX` syntaxes is incorrectly inferred, reported as `Any` + `(type alias)`.

**To Reproduce**
```py
x = 2**0b1010  # `(type alias) x: Any`
```

**Expected behavior**
`x` should appear either as an evaluated `Literal` (here, `Literal[1024]`), or `int` like `2**10` is.

**Screenshots or Code**
![Screenshot from Visual Studio Code showing the code from "To Reproduce" section, with `x` being hovered, revealing a box saying "(type alias) x: Any"](https://user-images.githubusercontent.com/43090614/223965251-16427efe-0c0e-402b-812c-954279f4e424.png)

**VS Code extension ~~or command-line~~**
Pylance 2023.3.20 (pyright 93a4ef87)

**Additional context**
I encountered the issue by assigning the aforementioned value to a constant, resulting in a similar behavior, except that instead of having `(type alias)`, it correctly shows `(constant)`.

**Comment by qexat** [comment]
While digging into the issue, I encountered a bug that is definitely related:

The following works fine:

```py
x = 0b1010  # (variable) x: Literal[10]
```

but adding manually the annotation throws an error, apparently:

```py
x: Literal[10] = 0b1010  # Expression of type "Literal[10]" cannot be assigned to declared type "Literal[10]" "Literal[10]" cannot be assigned to type "Literal[10]"
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.299, which I just published. It will also be included in a future release of pylance.
